tests/data/test_adjustments.py

```diff
@@ -167,6 +175,10 @@ def T(n):
             "Couldn't compute ratio for dividend sid=2, ex_date=1990-10-18,"
             " amount=10.000",
         ))
+        self.assertTrue(self.log_handler.has_warning(
```

Comment:
For paranoia's sake, can we also check that we handle the case where this is well beyond the end of the pricing data (e.g., 10 days or something)? Since we're fiddling with indices in the code under test here, I could imagine a world where this worked for dates one past the end of the array, but failed if we were well after the end.

Reply:
Might also be worth checking that we handle an ex_date that's before our inputs.

Reply:
added these cases